packages/react-dom/src/events/SyntheticMouseEvent.js

```diff
@@ -34,6 +37,24 @@ const SyntheticMouseEvent = SyntheticUIEvent.extend({
         : event.fromElement)
     );
   },
+  movementX: function(event) {
+    if ('movementX' in event) {
+      return event.movementX;
+    }
+
+    const screenX = previousScreenX;
+    previousScreenX = event.screenX;
+    return screenX ? event.screenX - screenX : 0;
```

Comment:
I think the check here intends to check against `null` but will accidentally also be falsy if `screenX` is actually equal to `0`. Please avoid truthy checks like this, they tend to hide bugs.

Reply:
If it's equal to `0` then it will still return `0` because that's the second expression in the ternary.

Reply:
I think that if saved `screenX` is `0`, but then you moved to `200`, you want to return `200 - 0` rather than `0`.

Reply:
I believe Dan is right, and we need to correct this.

Reply:
https://github.com/tc39/proposal-nullish-coalescing would have helped a lot with this, yes we should check if equal to null

Reply:
Another thing is that using the same variable for both `null` and numeric values can mess with VM optimizations. I think it would be better to use a separate boolean flag here.

Reply:
Also, are we happy to assume screenX on event? If undefined we could end up with `undefined - null`

Reply:
Yep, [screenX/Y are very well supported](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/screenX#Browser_compatibility).

Reply:
Ok so if screenX will always be an Number, could we not just initiate movementX as 0 Instead of null, and then instead of a ternary or null checking, always perform the minus operation. Because we can guarantee both operands will be Numbers by this point . I could be way off here but that sounds like it would work?

Reply:
Wouldn't make the first `movementX` equal to the current screen coordinate, instead of `0`? I think 0 would be expected (because the cursor didn't "jump" from top left point).